pysteps/downscaling/rainfarm.py

```diff
@@ -44,7 +44,60 @@ def _balanced_spatial_average(x, k):
     return convolve(x, k) / convolve(ones, k)
 
 
-def downscale(precip, ds_factor, alpha=None, threshold=None, return_alpha=False):
+def _smoothconv(P,nas):
```

Comment:
We are trying to enforce python [naming conventions](https://realpython.com/python-pep8/#naming-styles) for variables names. In this case, instead of `P` we could try to use a more informative name and apply the lower_with_under case. It is a generally a good idea to have a look at the existing code: we used `precip` as variable name for the input field.